src/domain/products/product-form/sections/sales-channels/product-availability-modal/add-channels-modal-screen.tsx

```diff
@@ -0,0 +1,130 @@
+import { SalesChannel } from "@medusajs/medusa"
+import { useAdminSalesChannels } from "medusa-react"
+import React, { useState } from "react"
+import { usePagination, useRowSelect, useTable } from "react-table"
+import Button from "../../../../../../components/fundamentals/button"
+import Modal from "../../../../../../components/molecules/modal"
+import { LayeredModalContext } from "../../../../../../components/molecules/modal/layered-modal"
+
+import SalesChannelAvailabilityTable, {
+  useAvailableChannelsModalTableColumns,
+} from "./sales-channel-availability-table"
+
+export const useAddChannelsModalScreen = (
+  selectedRowIds,
+  onAddSelectedToAvailableChannels
+) => {
+  const { pop } = React.useContext(LayeredModalContext)
+
+  return {
+    title: "Add Sales Channels",
+    onBack: pop,
+    view: (
+      <AddChannelsModalScreen
+        selectedSalesChannelIds={selectedRowIds}
+        onAddSelectedToAvailableChannels={onAddSelectedToAvailableChannels}
+      />
+    ),
+  }
+}
+
+type AddChannelsModalScreenProps = {
+  selectedSalesChannelIds: string[]
+  onAddSelectedToAvailableChannels: (salesChannels: SalesChannel[]) => void
+}
+
+const LIMIT = 15
+
+const AddChannelsModalScreen: React.FC<AddChannelsModalScreenProps> = ({
+  onAddSelectedToAvailableChannels: onAddSelectedToAvailableChannels,
+  selectedSalesChannelIds,
+}) => {
+  const [offset, setOffset] = useState(0)
+  const [query, setQuery] = useState("")
+  const [freeText, setFreeText] = useState("")
+
+  const { pop } = React.useContext(LayeredModalContext)
+  const [columns] = useAvailableChannelsModalTableColumns()
+
+  const { sales_channels: salesChannels } = useAdminSalesChannels({
+    q: freeText,
+  })
+
+  const numPages = Math.ceil((salesChannels?.length || 0) / LIMIT)
```

Comment:
issue: this pagination won't work properly if there is > 15 channels
(also the hook will always fetch only the first 20 sales channels)

Reply:
Fixed the pagination 😄